jwst/flatfield/flat_field_step.py

```diff
@@ -63,6 +63,16 @@ def process(self, input):
 
         # Retrieve the reference file name or names
         if is_NIRSpec:
+            exp_type = input_model.meta.exposure.type
+            if exp_type not in ["NRS_BRIGHTOBJ", "NRS_FIXEDSLIT",
+                                "NRS_IFU", "NRS_MSASPEC"]:
```

Comment:
There is currently no flat-field reference file for "NRS_LAMP", so if we included that in the list of allowed exposure types, the step would fail at the point of trying to get the reference file.  We don't want the step to fail, we just want to set S_FLAT to "SKIPPED" and return the input.